Modules/modFileEncoding.ACM

```diff
@@ -41,6 +76,105 @@ Function IsUTF8(strFilename As String) As Boolean
   End If
 End Function
 
+Function ConvFile_to_UCS2(strFilename As String, Optional strNewFilename As String) As Boolean
+  ' Converts the supplied file from any of ANSI, UTF-8 or UCS2 into UCS2 format
+  
+  ConvFile_to_UCS2 = False ' initially set to false
+
+  If LenB(strNewFilename) = 0 Then strNewFilename = strFilename
+
+  ' deal with case where no conversino is required
+  If IsUCS2(strFilename) Then
+    If strFilename = strNewFilename Then
+      ' do nothing the file is already in UCS2 format
+      ConvFile_to_UCS2 = True
+    Else
+      ' copy file to new location without conversion
+      If FileExists(strNewFilename) Then Kill strNewFilename
+      FileCopy strFilename, strNewFilename
+      ConvFile_to_UCS2 = True
+    End If
+    Exit Function
+  End If
+  
+  ' we need to do a conversion
+  If IsUTF8(strFilename) Then
+    ConvFile_to_UCS2 = ConvFileUTF8_to_UCS2(strFilename, strNewFilename)
+  ElseIf IsANSI(strFilename) Then
+    ConvFile_to_UCS2 = ConvFileANSI_to_UCS2(strFilename, strNewFilename)
+  End If
+  
+End Function
+
+Function ConvFileANSI_to_UCS2(strFilename As String, Optional strNewFilename As String) As Boolean
+  ' Converts the supplied file from ANSI (8 bit) to UCS2 format
+  
+  Dim byts() As Byte
+  Dim bytsOut() As Byte
+  Dim lngLen As Long
+  Dim lngOutLen As Long
+  Dim i As Long
+  Dim j As Long
+    
+  ConvFileANSI_to_UCS2 = False ' initially set to false
+
+  If LenB(strNewFilename) = 0 Then strNewFilename = strFilename
+
+  If ReadFromFile(strFilename, byts) Then
+    lngLen = UBound(byts) + 1
+    lngOutLen = lngLen * 2 + 2 ' 2 bytes for every ANSI byte plus 2 Byte BOM
+    ReDim bytsOut(lngOutLen - 1)
+    ' Insert BOM
+    bytsOut(0) = 255 ' &HFF
+    bytsOut(1) = 254 ' &HFE
+    ' j is the position in the output array
+    j = 2
+    For i = 0 To UBound(byts)
+      bytsOut(j) = byts(i)
+      j = j + 1
+      bytsOut(j) = 0
+      j = j + 1
+    Next i
+    ConvFileANSI_to_UCS2 = WriteToFile(bytsOut, strNewFilename)
+  End If
+  
+End Function
+
+Function ConvFileUCS2_to_ANSI(strFilename As String, Optional strNewFilename As String) As Boolean
+  ' Converts the supplied file from UCS2 to ANSI format - IF IT CAN!
+  ' NB Depending on the file contents it may or may not be possible to actually do this conversion
+  
+  Dim byts() As Byte
+  Dim bytsOut() As Byte
+  Dim lngOutLen As Long
+  Dim i As Long
+  Dim j As Long
+  
+  ConvFileUCS2_to_ANSI = False ' initially set to false
+  
+  If LenB(strNewFilename) = 0 Then strNewFilename = strFilename
+  
+  ' load the file into the bytsFile array
+  If ReadFromFile(strFilename, byts) Then
+    If byts(LBound(byts)) = 255 And byts(LBound(byts) + 1) = 254 Then  ' 255 = &HFF, 254 = &HFE (the BOM)
+      For i = 3 To UBound(byts) Step 2

```

Comment:
should be `i = LBound(byts) + 2` ?

Reply:
I have changed it to LBound(byts) + 3 which is correct as we are looking for the zeros in this loop and it is little endian.